tests/simulator/test_initialization.py

```diff
@@ -0,0 +1,587 @@
+                                 compute_timestamps=np.zeros(1),
+                                 population_name=pop)
+
+            for quantity in ['domain', 'levelGhost', 'patchGhost']:
+                ParticleDiagnostics(quantity=quantity,
+                                    compute_timestamps=np.zeros(1),
+                                    write_timestamps=np.zeros(1),
+                                    population_name=pop)
+
+        simulator = Simulator(globals.sim)
+        simulator.initialize()
+        simulator.diagnostics().dump(timestamp=0, timestep=1)
+
+        if qty == "b":
+            b_hier = hierarchy_from(h5_filename=diag_outputs+"/EM_B.h5")
+            return b_hier
+
+
+        if qty == "particles":
+            particle_hier = hierarchy_from(h5_filename=diag_outputs+"/ions_pop_protons_domain.h5")
+            particle_hier = hierarchy_from(h5_filename=diag_outputs+"/ions_pop_protons_patchGhost.h5", hier=particle_hier)
+            particle_hier = hierarchy_from(h5_filename=diag_outputs+"/ions_pop_protons_levelGhost.h5", hier=particle_hier)
+            merge_particles(particle_hier)
+            return particle_hier
+
+        if qty == "moments":
+            mom_hier = hierarchy_from(h5_filename=diag_outputs+"/ions_density.h5")
+            mom_hier = hierarchy_from(h5_filename=diag_outputs+"/ions_bulkVelocity.h5", hier=mom_hier)
+            mom_hier = hierarchy_from(h5_filename=diag_outputs+"/ions_pop_protons_density.h5", hier=mom_hier)
+            mom_hier = hierarchy_from(h5_filename=diag_outputs+"/ions_pop_protons_flux.h5", hier=mom_hier)
+            if beam:
+                mom_hier = hierarchy_from(h5_filename=diag_outputs+"/ions_pop_beam_density.h5", hier=mom_hier)
+                mom_hier = hierarchy_from(h5_filename=diag_outputs+"/ions_pop_beam_flux.h5", hier=mom_hier)
+            return mom_hier
+
+
+
+
+
+    @data(1,2,3)
+    def test_B_is_as_provided_by_user(self, interp_order):
+        print("test_B_is_as_provided_by_user : interp_order : {}".format(interp_order))
+        hier = self.getHierarchy(interp_order, {"L0": {"B0": [(10, ), (20, )]}}, "b")
+
+        from pyphare.pharein import globals
+        model = globals.sim.model
+        bx_fn = model.model_dict["bx"]
+        by_fn = model.model_dict["by"]
+        bz_fn = model.model_dict["bz"]
+
+        for ilvl, level in hier.levels().items():
+            print("checking level {}".format(ilvl))
+            for ip, patch in enumerate(level.patches):
+
+                xbx   = patch.patch_datas["EM_B_x"].x[:]
+                bx  = patch.patch_datas["EM_B_x"].dataset[:]
+                np.testing.assert_allclose(bx, bx_fn(xbx), atol=3e-5)
+
+                xby   = patch.patch_datas["EM_B_y"].x[:]
+                by  = patch.patch_datas["EM_B_y"].dataset[:]
+                np.testing.assert_allclose(by, by_fn(xby), atol=3e-5)
+
+                xbz   = patch.patch_datas["EM_B_z"].x[:]
```

Comment:
not directly related to this PR (I think), but should this
```
https://github.com/nicolasaunai/PHARE/blob/ghosttest/pyphare/pyphare/pharesee/hierarchy.py#L58
```

be a property?

Reply:
maybe, but we do not necessarily want to compute that each time